roles/rhel_subscribe/tasks/main.yml

```diff
@@ -30,7 +30,14 @@
   redhat_subscription:
     username: "{{ rhel_subscription_user }}"
     password: "{{ rhel_subscription_pass }}"
-    autosubscribe: yes
+
+- name: Retrieve the OpenShift Pool ID
+  command: subscription-manager list --available --matches="OpenShift Enterprise, Premium*" --pool-only
```

Comment:
@detiber: I have just added an option to configure the Pool name to attach to: `rhel_subscription_pool`. Its default value is `OpenShift Enterprise, Premium*`.

I wish I could have used a more complex regex like `OpenShift Enterprise, (Standard|Premium).*` but it seems the `--matched` switch of `subscription_manager` doesn’t support such regex.